maven-plugin/src/main/java/net/revelc/code/formatter/FormatterMojo.java

```diff
@@ -94,7 +94,7 @@
 	/**
 	 * Project's source directory as specified in the POM.
 	 */
-	@Parameter(defaultValue = "${project.build.sourceDirectory}", property = "sourceDirectory", required = true)
+	@Parameter(defaultValue = "${project.build.sourceDirectory}", readonly = true, required = true)
```

Comment:
this may not be true if you wanna the formatter to be run only once in a multi module project.

That is why I left this configurable.